hazelcast/src/main/java/com/hazelcast/query/impl/getters/ExtractorGetter.java

```diff
@@ -42,7 +43,10 @@ Object getValue(Object target) throws Exception {
         // This part will be improved in 3.7 to avoid extra allocation
         DefaultValueCollector collector = new DefaultValueCollector();
         if (target instanceof Data) {
-            extractionTarget = new GenericRecordQueryReader(serializationService.readAsInternalGenericRecord((Data) target));
+            InternalGenericRecord internalGenericRecord = serializationService.readAsInternalGenericRecord((Data) target);
```

Comment:
How come this can be null now? I see that `readAsInternalGenericRecord` documentation says it returns null if it cannot be represented with generic record, but on the implementation we throw, not return null

Reply:
Removing the null check and I am also updating the javadoc

Reply:
Fixed by 45694b9